test/unit/server-ha-statefulset.bats

```diff
@@ -157,6 +157,28 @@ load _helpers
   [ "${actual}" = "10" ]
 }
 
+@test "server/ha-StatefulSet: custom replicas" {
+  cd `chart_dir`
+  local actual=$(helm template \
+      --show-only templates/server-statefulset.yaml  \
+      --set 'server.ha.enabled=true' \
+      --set 'server.ha.replicas=0' \
+      . | tee /dev/stderr |
+      yq -r '.spec.replicas' | tee /dev/stderr)
+  [ "${actual}" = "0" ]
+}
+
+@test "server/ha-StatefulSet: custom replicas" {
+  cd `chart_dir`
+  local actual=$(helm template \
+      --show-only templates/server-statefulset.yaml  \
+      --set 'server.ha.enabled=true' \
+      --set 'server.ha.replicas=null' \
```

Comment:
Nice tests! I'm 99% sure it will work, but just out of paranoia could we also add a test case where we don't set `server.ha.replicas` at all?

Reply:
I'm not sure how we'd go about that, as `server.ha.replicas` is always set through the default `values.yaml` file. There is already a test case in the `server-ha-statefulset.bats` file for _no explicit value_, so I'm not sure what else to cover with the unit test. We have the following options now:

- No explicit value -> 3
- Arbitrary, non zero value -> 10
- Zero -> 0
- `null` -> 3

What other case would you add?

Reply:
Oh my bad, I forgot it's set in the `values.yaml` file. Weird that the templating code previously used ` | default 3` in that case 🤔 but feel free to ignore this comment!